apps/website/src/server/db/show-and-tell.ts

```diff
@@ -27,6 +27,25 @@ export const whereApproved = {
   approvedAt: { gte: prisma.showAndTellEntry.fields.updatedAt },
 };
 
+function getPostFilter(filter: "approved" | "pendingApproval") {
+  return filter === "pendingApproval"
+    ? {
+        OR: [
+          { approvedAt: null },
+          { approvedAt: { lt: prisma.showAndTellEntry.fields.updatedAt } },
+        ],
+      }
+    : filter === "approved"
+    ? { approvedAt: { gte: prisma.showAndTellEntry.fields.updatedAt } }
+    : {};
+}
+
+const postOrderBy = [
+  { seenOnStreamAt: { sort: "desc", nulls: "first" } },
```

Comment:
Is there a way to sort just by whether this is null or not, rather than by the value in the column? It think its a bit confusing to have the posts that've been seen be stored by when they were seen, as that's not the date shown?

Reply:
I did not find any way. Maybe we could manage our own sort value as a separate column. The only way to do this would probably be not using prisma (or using a raw query). They sadly do not support SQL syntax for the order by https://github.com/prisma/prisma/issues/5848

For now I would just try clicking "seen" in the right order...

Reply:
Ah, that's a shame -- I guess the other solution would be to make the column a boolean instead of a nullable timestamp?

Reply:
that should work! i would not mind that

Reply:
i would just add it as a second col though

Reply:
Works for me :)